add registration of VC validators back in the new per validator management of builders

stephanep opened the pull request:
Fix handling of VC validators by per validator payload-builder registration (issue #5599).

The proposed change ensures that VC validators are registered with the builder specified by the `--payload-builder-url` argument even if the beacon node has no attached validators. It also prevent such validators from being unintentionally registered with builders configured for specific attached validators by the keymanager api.

tersec commented on beacon_chain/validators/beacon_validators.nim:
```diff
@@ -1576,7 +1576,8 @@ proc registerValidatorsPerBuilder(
     var nonExitedVcPubkeys: HashSet[ValidatorPubKey]
-    if node.externalBuilderRegistrations.len > 0:
+    if  node.externalBuilderRegistrations.len > 0 and
+        payloadBuilderAddress == node.config.getPayloadBuilderAddress.value:
```
`node.config.getPayloadBuilderAddress` is a BN fallback default, which probably should not be showing up in `registerValidatorsPerBuilder`, except for the discussion at https://github.com/status-im/nimbus-eth2/issues/5599#issuecomment-1812894337 and on its face looks like a bug, rather than something intentional; perhaps worth copying some of the analysis from that issue comment into the source code, where it won't be lost.

spoignant86 replied:
ok, added to the commit

stephanep committed:
Fix handling of VC validators by per payload-builder payload builder validator registration (issue #5599).
The proposed change ensures that VC validators are registered with the builder specified by the `--payload-builder-url` argument even if the beacon node has no attached validators. It also prevent such validators from being unintentionally registered with builders configured for specific attached validators by the keymanager api.
Per the current specs, the VC have no way to specify which builders the BN should use on a per-node basis, so for the time being we have to resort to using the BN fallback default builder URL for VC validators.
